src/main/java/edu/ucdavis/dss/ipa/entities/InstructorNote.java

```diff
@@ -0,0 +1,98 @@
+package edu.ucdavis.dss.ipa.entities;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+import javax.persistence.Transient;
+import javax.validation.constraints.NotNull;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import edu.ucdavis.dss.ipa.api.deserializers.InstructorNoteDeserializer;
+
+@SuppressWarnings("serial")
+@Entity
+@Table(name = "InstructorNotes")
+@JsonDeserialize(using = InstructorNoteDeserializer.class)
+@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
+public class InstructorNote extends BaseEntity {
+  private long id;
+  private String note;
+  private Instructor instructor;
+  private Schedule schedule;
+
+  @Id
+  @GeneratedValue(strategy = GenerationType.IDENTITY)
+  @Column(name = "Id", unique = true, nullable = false)
+  @JsonProperty
+  public long getId()
+  {
+    return this.id;
+  }
+
+  public void setId(long id)
+  {
+    this.id = id;
+  }
+
+  @ManyToOne(fetch = FetchType.LAZY)
+  @JoinColumn(name = "ScheduleId", nullable = false)
+  @NotNull
+  @JsonIgnore
+  public Schedule getSchedule() {
+    return schedule;
+  }
+
+  public void setSchedule(Schedule schedule) {
+    this.schedule = schedule;
+  }
+
+  @ManyToOne(fetch = FetchType.LAZY)
+  @JoinColumn(name = "InstructorId", nullable = false)
+  @NotNull
+  @JsonIgnore
+  public Instructor getInstructor() {
+    return instructor;
+  }
+
+  public void setInstructor(Instructor instructor) {
+    this.instructor = instructor;
+  }
+
+  @JsonProperty
+  public String getNote() {
+    return note;
+  }
+
+  public void setNote(String note) {
+    this.note = note;
+  }
+
+  @JsonProperty("instructorId")
+  @Transient
+  public long getInstructorIdentification() {
```

Comment:
Why is this getter and the one below it called "Identification" instead of Id?

Reply:
We started following this pattern quite a while back, originally we did use getInstructorId() on transient methods like this one,
but I vaguely recall that after much debugging we found that names like getInstructorId() caused Hibernate to incorrectly infer things about the entity and it caused problems during repository actions. A slight change to the name so it no longer matched hibernate syntax but still made sense to humans seemed straightforward.